lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1546,7 +1569,10 @@ static bool ParseIRGenArgs(IRGenOptions &Opts, ArgList &Args,
   } else if (const Arg *A = Args.getLastArg(OPT_g_Group)) {
     if (A->getOption().matches(OPT_g))
       Opts.DebugInfoKind = IRGenDebugInfoKind::Normal;
-    else if (A->getOption().matches(options::OPT_gline_tables_only))
+    else if (A->getOption().matches(options::OPT_gline_tables_only) ||
+             // Requesting optimization remarks implies source line debug info
+             // in order to tie the optimization back to the source line.
+             Args.hasArg(OPT_Rpass_EQ) || Args.hasArg(OPT_Rpass_missed_EQ))
```

Comment:
This is unfortunate. Is it possible to get it stripped out before emitting the .o file?

Reply:
I don't think we actually need this piece until we hook this up to LLVM.  This patch does yet forward -Rpass to LLVM.

In the LLVM passes we only get source location with source-line debug info.  Looks like that in SIL they are always present and then it's left to IRGen whether to generate LLVM debug info.